universal-application-tool-0.0.1/app/views/LoginForm.java

```diff
@@ -19,12 +19,28 @@ public LoginForm(BaseHtmlLayout layout) {
   }
 
   public Content render(Http.Request request, Optional<String> message) {
-    return layout.htmlContent(
+    ContainerTag bodyTag =
         body(
-            h1("Error: You are not logged in").withCondHidden(!message.orElse("").equals("login")),
-            h1("TODO: IDCS integration"),
-            h1("Or, continue as guest."),
-            redirectButton(
-                "guest", "Continue", routes.CallbackController.callback("GuestClient").url())));
+            div(
+                h1("Error: You are not logged in")
+                    .withCondHidden(!message.orElse("").equals("login")),
+                h1("TODO: IDCS integration")),
+            div(
+                h1("Or, continue as guest."),
+                redirectButton(
+                    "guest", "Continue", routes.CallbackController.callback("GuestClient").url())));
+    // "defense in depth", sort of - this client won't be present in production, and this button
```

Comment:
Question: What does "defense in depth" mean? 😃  Meaning it's secure because there are sort of two levels of things this should be blocked on in prod?

Reply:
Yep - the "swiss cheese model".  If you have a bunch of layers that are full of holes, you can still be secure as long as the holes don't all line up.  :p